File: EIMS/EIMS_Lib/ViewModels/HolidayViewModel.cs
using EIMS_Lib.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EIMS_Lib.ViewModels
{
    public class HolidayViewModel
    {
        public int HolidayId { get; set; }
        
        public DateTime HolidayDate  { get; set; }
      
        public HolidayType HolidayType { get; set; }
        
        public DateTime Upto { get; set; }
       
        public string Description { get; set; }
    }
}

File: EIMS/EIMS_UI/ClientApp/src/app/models/view-models/holiday-view-model.ts
import { HolidayType } from '../common/constants-model';

export class HolidayViewModel {
  constructor(
    public holidayId?: number,
    public holidayDate?: string,
    public holidayType?: HolidayType,
    public upto?: Date | string,
    public description?: string
  ) { }
}
